apps/hash-frontend/src/pages/shared/entities-table.tsx

```diff
@@ -45,23 +45,37 @@ import {
   useEntitiesTable,
 } from "./entities-table/use-entities-table";
 import { TOP_CONTEXT_BAR_HEIGHT } from "./top-context-bar";
-import { WorkspaceContext } from "./workspace-context";
 
 export const EntitiesTable: FunctionComponent<{
   hideEntityTypeVersionColumn?: boolean;
   hidePropertiesColumns?: boolean;
 }> = ({ hideEntityTypeVersionColumn, hidePropertiesColumns }) => {
   const router = useRouter();
 
-  const { activeWorkspaceOwnedById } = useContext(WorkspaceContext);
+  const { authenticatedUser } = useAuthenticatedUser();
 
   const [filterState, setFilterState] = useState<FilterState>({
     includeGlobal: false,
   });
   const [showSearch, setShowSearch] = useState<boolean>(false);
 
-  const { entities, entityTypes, propertyTypes, subgraph } =
-    useEntityTypeEntities();
+  const {
+    entities: lastLoadedEntities,
+    entityTypes,
+    hadCachedContent,
+    loading,
+    propertyTypes,
+    subgraph,
+  } = useEntityTypeEntities();
+
+  const entities = useMemo(
+    /**
+     * If a network request is in process and there is no cached content for the request, return undefined.
+     * There may be stale data in the context related to an earlier request with different variables.
+     */
+    () => (loading && !hadCachedContent ? undefined : lastLoadedEntities),
+    [hadCachedContent, loading, lastLoadedEntities],
+  );
```

Comment:
Is this any different than just using the current value for `entities?

```suggestion
  const entities = lastLoadedEntities;
```

I'm wondering because:
- when there is no cached content and it is loading, `lastLoadedEntities` will be `undefined`
- when there is cached content and it is loading, `lastLoadedEntities` will be defined and updated when the latest fetch has resolved
- when there it is not loading and there is cached content, `lasLoadedEntities` will be defined

Reply:
Ah apologies, I think I missed that this wouldn't account for stale data being displayed.

Would it perhaps be simpler to just move this logic into the `useEntityTypeEntitiesContextValue` hook in a way that prevents stale data and only cached data for the correct query variables being returned in the `entities` array? And otherwise to just return `undefined` whilst it's loading? This could help the the hook callers from having to make use of `hadCachedContent`

Reply:
There might be contexts in which it is helpful to still have the old data visible while a variable is changed... e.g. if moving to the next page of a table, or changing the sort, it might be slightly less jarring to keep the old rows in place (with some loading indicator) than to wipe the rows while the query resolved. 

This might not happen in practice but I have a slight preference for keeping it how it is and changing it later depending on how else we use it.

Reply:
Alright, fair enough. Thanks for explaining